src/utils/create-broadcast.js

```diff
@@ -7,29 +7,36 @@
 
 export type Broadcast = {
   publish: (value: mixed) => void,
-  subscribe: (listener: (currentValue: mixed) => void) => () => void
+  subscribe: (listener: (currentValue: mixed) => void) => number,
+  unsubscribe: (number) => void,
 }
 
-const createBroadcast = (initialValue: mixed): Broadcast => {
-  let listeners = []
-  let currentValue = initialValue
+const createBroadcast = (initialState: mixed): Broadcast => {
+  const listeners = {}
+  let id = 0
+  let state = initialState
 
-  return {
-    publish(value: mixed) {
-      currentValue = value
-      listeners.forEach(listener => listener(currentValue))
-    },
-    subscribe(listener) {
-      listeners.push(listener)
+  const publish = (nextState: mixed) => {
+    state = nextState
+    const listenerKeys = Object.keys(listeners)
+    for (let i = 0; i < listenerKeys.length; i += 1) {
+      listeners[listenerKeys[i]](state)
+    }
+  }
 
-      // Publish to this subscriber once immediately.
-      listener(currentValue)
+  const subscribe = listener => {
+    const currentId = id
+    listeners[currentId] = listener
+    id += 1
+    listener(state)
+    return currentId
+  }
 
-      return () => {
-        listeners = listeners.filter(item => item !== listener)
-      }
-    },
+  const unsubscribe = unsubID => {
+    delete listeners[unsubID]
```

Comment:
Minimal I'd expect since it's a simple Boolean guard. if `(x === undefined) { continue; }` should be fine and lightweight